applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,741 @@
+		return true;
+	default:
+		return false;
+	}
+}
+
+static void enable_qos_reporting(void)
+{
+	int err;
+	struct net_buf *buf;
+
+	err = bt_hci_register_vnd_evt_cb(on_vs_evt);
+	if (err) {
+		LOG_ERR("Failed to register HCI VS callback");
+		return;
+	}
+
+	hci_vs_cmd_qos_conn_event_report_enable_t *cmd_enable;
+
+	buf = bt_hci_cmd_create(HCI_VS_OPCODE_CMD_QOS_CONN_EVENT_REPORT_ENABLE,
+							sizeof(*cmd_enable));
+	if (!buf) {
+		LOG_ERR("Failed to enable HCI VS QoS");
+		return;
+	}
+
+	cmd_enable = net_buf_add(buf, sizeof(*cmd_enable));
+	cmd_enable->enable = 1;
+
+	err = bt_hci_cmd_send(
+			HCI_VS_OPCODE_CMD_QOS_CONN_EVENT_REPORT_ENABLE,
+			buf);
+	if (err) {
+		LOG_ERR("Failed to enable HCI VS QoS");
+		return;
+	}
+}
+
+static void handle_config_event(const struct config_event *event)
+{
+	switch (OPT_FIELD_GET(event->id)) {
+	case QOS_OPT_BLACKLIST:
+		__ASSERT_NO_MSG(
+			event->dyndata.size == sizeof(struct params_blacklist));
+		update_blacklist(event->dyndata.data);
+		break;
+	case QOS_OPT_CHMAP:
+		/* Avoid updating channel map directly to */
+		/* reduce complexity in interaction with */
+		/* chmap filter lib */
+		LOG_DBG("Not supported");
+		break;
+	case QOS_OPT_PARAM_BLE:
+		__ASSERT_NO_MSG(
+			event->dyndata.size == sizeof(struct params_ble));
+		update_parameters(event->dyndata.data, NULL);
+		break;
+	case QOS_OPT_PARAM_WIFI:
+		__ASSERT_NO_MSG(
+			event->dyndata.size == sizeof(struct params_wifi));
+		update_parameters(NULL, event->dyndata.data);
+		break;
+	default:
+		LOG_WRN("Unknown opt");
+		return;
+	}
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE)) {
+		if (IS_ENABLED(CONFIG_DESKTOP_HID_MOUSE)) {
+			static s32_t hid_pkt_recv_count;
+			static s32_t usb_pkt_sent_count;
+			static u32_t cdc_notify_count;
+
+			/* Count number of HID packets received via BLE */
+			/* and number of HID packets sent via USB. */
+			/* Send stats printout via CDC every 100 packets */
+
+			if (is_hid_mouse_event(eh)) {
+				hid_pkt_recv_count++;
+				cdc_notify_count++;
+
+				if (cdc_notify_count == 100) {
+					hid_pkt_stats_print(
+						hid_pkt_recv_count,
+						usb_pkt_sent_count);
+					cdc_notify_count = 0;
+				}
+
+				return false;
+			}
+
+			if (is_hid_report_sent_event(eh)) {
+				usb_pkt_sent_count++;
+
+				return false;
+			}
+		}
+	}
+
+	if (is_module_state_event(eh)) {
+		const struct module_state_event *event =
+			cast_module_state_event(eh);
+
+		if (check_state(
+				event,
+				MODULE_ID(main),
+				MODULE_STATE_READY)) {
+			static bool initialized;
+
+			__ASSERT_NO_MSG(!initialized);
+
+			chmap_inst = chmap_filter_init();
+			__ASSERT_NO_MSG(chmap_inst != NULL);
+
+			k_mutex_init(&data_access_mutex);
+			new_params = NULL;
+			new_blacklist = INVALID_BLACKLIST;
+
+			LOG_DBG("Chmap lib version: %s",
+				chmap_filter_version());
+			enable_qos_reporting();
+
+#if CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE
```

Comment:
ping

Reply:
CONFIG_USB_CDC_ACM_DEVICE_NAME is not defined when CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE = 0. Would have to add #ifdef for CONFIG_USB_CDC_ACM_DEVICE_NAME to avoid compilation error

Reply:
you can add following at the file's head
```
#ifdef CONFIG_USB_CDC_ACM_DEVICE_NAME 
   #define USB_SERIAL_DEVICE_NAME CONFIG_USB_CDC_ACM_DEVICE_NAME 
#else 
   #define USB_SERIAL_DEVICE_NAME  NULL
#endif 
```
and later use `USB_SERIAL_DEVICE_NAME`

using `if (IS_ENABLED)` provides free source checking regardless of the configuration. Also makes code consistent.